### PR TITLE
[Response required]Add support for web identity provider in AWS

### LINKED ISSUE #19393: [Metricbeat] Add support for IAM Roles for Service Accounts (EKS)
**Describe the enhancement:**

It would be really nice to be able to use this additional authentication mechanism when running metricbeat inside Elastic Kubernetes Service (specially when using the [AWS Module](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-aws.html)).

**Describe a specific use case for the enhancement or feature:**

I tried to configure metricbeat following the AWS official documentation [Specifying an IAM role for your service account](https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html) (we use this approach successfully with other components like ALB Ingress Controller, External DNS, Fluxcd and so on) and even if I change the container permissions to be able to read the token file (fsGroup: 472) metricbeat insists to use the worker node role (which is not what we want).

We are using metricbeat version 7.8.0.

This feature would enable more flexibility and security since metricbeat requires some additional permissions that may not be desirable to share with all pods running on the same cluster.

Follow the configuration we tried to use on metricbeat deployment (setting fsGroup to 472 works for other deployments):

```yaml
securityContext:
  fsGroup: 472
  runAsGroup: 472
  runAsUser: 0
```
We don't manually specify any AWS auth related env because AWS does that for us (when using IRSA) and with that the AWS SDK should detect the injected environments (AWS_ROLE_ARN and AWS_WEB_IDENTITY_TOKEN_FILE).

Thanks.

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

WebIdentityRoleProvider is used to retrieve credentials using an Open ID Connect Providers(OIDC) token. This feature allows us to authenticate AWS API calls with supported identity providers and receive a valid OIDC JSON web token (JWT). You can pass this token to the AWS STS AssumeRoleWithWebIdentity API operation and receive IAM temporary role credentials. 

This PR is to add support for using web identity role provider in Beats for AWS.

## Why is it important?

AWS EKS starts to support an [IAM roles for service account (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to provide temporary credentials and AWS permissions to containers in any pod that uses this service account.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues
#19393

### CODE REVIEW

File: x-pack/libbeat/common/aws/credentials.go [hunk 1]
```diff
@@ -67,6 +74,27 @@ func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
...
+	emptyWebIdentity := WebIdentityProvider{}
+	if config.WebIdentityProvider != emptyWebIdentity {
+		logger.Debug("Using web_identity_role for AWS credential")
+		awsConfig, err := external.LoadDefaultAWSConfig()
+		if err != nil {
+			return awsConfig, errors.Wrap(err, "external.LoadDefaultAWSConfig failed when using role_arn")
+		}
+
+		stsSvc := sts.New(awsConfig)
+		p := stscreds.NewWebIdentityRoleProvider(stsSvc, config.WebIdentityProvider.RoleArn, config.WebIdentityProvider.RoleSession, stscreds.IdentityTokenFile(config.WebIdentityProvider.TokenFilePath))
```
Comment (sboulkour) [inline x-pack/libbeat/common/aws/credentials.go:86]: @kaiyan-sheng is the `RoleSession` param mandatory ? Just being curious as I activated the feature on my Filebeat pod, and only `RoleArn` + `TokenFilePath` seemed to be passed to the pod by EKS :
```
[...]
    - name: AWS_ROLE_ARN
      value: arn:aws:iam::<HIDDEN>:role/<HIDDEN>
    - name: AWS_WEB_IDENTITY_TOKEN_FILE
      value: /var/run/secrets/eks.amazonaws.com/serviceaccount/token
[...]
```

**Comment by martijnbakkerelastic** [comment]
Requesting for feedback on this PR. Customer is currently working with a temporary workaround (IAMUSER) but is awaiting outlook to use IAMROLE. Current workaround is under Security vulnerability-exception, but this will expire coming week and in order to request extension our customer Ikano Bank is requesting clarity on expected timelines when this is planned to be added. This will allow them to request an extension on the current Security exception in place.

It seems activities did start on creating the build H2 2020), but never completed. Can we have any insights on when/how this will be picked up again and what the expected ETA is?

The pending change prevents additional AWS sources to be incorporated in their service and is a blocker for further growth in their current implementation.

**Comment by martijnbakkerelastic** [comment]
Is there a way for someone with the eligible knowledge/mandate to provide me with an update on this pending request? As stated two weeks ago, we are trying to assist our customer in providing them an outlook on planned implementation so they will be abel to request an expanded vulnerability exception.

**Comment by kaiyan-sheng** [comment]
@martijnbakkerelastic Hi!! SORRY for the late response...... This work is not planed for 7.14 but I can add it into our 7.15 planning. Will that be OK? Thanks!!!!

**Comment by martijnbakkerelastic** [comment]
@kaiyan-sheng Thanks for picking this up again. I think the freeze date for 7.14 is end this month, so if we could embed it sooner that would be high preferable (the issue was already addressed back in October last year). I believe the release dates planned are early August for 7.14 and end September for 7.15 right?

**Comment by mtparet** [comment]
Hello,

We are in process of implementing a high secure environment and this is a must have.

Thank you for tackling this !

**Comment by kaiyan-sheng** [comment]
Closing this draft PR since I keep having a hard time putting this in my schedule. We have another engineer who just picked this up 😄 New PR: https://github.com/elastic/beats/pull/27126

**Pull request closed without merging**